Project2015To2017/Reading/ProjectReader.cs

```diff
@@ -1,28 +1,48 @@
+using Project2015To2017.Definition;
 using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.IO;
 using System.Linq;
 using System.Xml;
 using System.Xml.Linq;
-using Project2015To2017.Definition;
 using static Project2015To2017.Definition.Project;
 
 namespace Project2015To2017.Reading
 {
 	public class ProjectReader
 	{
-		public Project Read(string filePath)
+		public static readonly ProjectReader Instance = new ProjectReader();
```

Comment:
I think we should leave this an instance but then initialize it with either a `Project` or a `FileInfo` if we want to be able to reapply optimizations.
We can obsolete the other read method in that case to remain backwards compatible.

Reply:
I wasn't sure on the why of having the static field. I would keep it an instance still that you can initialize with either a file or a project definition. You could then by using obsolete attributes keep the existing contract so we're backwards compatible.

Or maybe I missed the intent of your change completely :smile:

Reply:
Ok, you mean removing Instance field and adding constructors to initialize some `FileInfo Project` field? But then there is no point in caching, unless cache is made static. Is that ok?